### ISSUE TITLE
Formatter removes semicolon after the virtual modifier declaration

### ISSUE THREAD

**Issue by matmilbury** [NomicFoundation/hardhat-vscode#219]
I have the following file:

```sol
// SPDX-License-Identifier: MIT
pragma solidity 0.8.2;

abstract contract Foo {
    modifier bar() virtual;
}

```

When I format it with `hardhat-vscode`, the formatter removes the semicolon after the modifier:
```diff
// SPDX-License-Identifier: MIT
pragma solidity 0.8.2;

abstract contract Foo {
-    modifier bar() virtual;
+    modifier bar() virtual
}

```

which results in invalid solidity code.

The issue seems to disappear when the modifier gets a default implementation:
```sol
// SPDX-License-Identifier: MIT
pragma solidity 0.8.2;

abstract contract Foo {
    modifier bar() virtual {
        _;
    }
}

```

**Comment by kanej** [comment]
I have been able to reproduce the example via the `hardhat-vscode` format call.

When I run the on the same input via `prettier-plugin-solidity` which we use for formatting under the hood, we don't get this issue.

The next step is to debug and figure out why formatting via the plugin gives a different result than via prettier at the cli.

**Comment by kanej** [comment]
Released as part of `v0.5.0`

**Comment by matmilbury** [comment]
Thank you! 🙇
